scripts/linux/setup.sh

```diff
@@ -0,0 +1,180 @@
+}
+
+get_deps_source_code() {
+    CLONE_DIRECTORY=$1
+    pushd "${CLONE_DIRECTORY}"
+
+    [ -d "glog" ] || {
+       git clone --branch v0.3.5 --depth 1 https://github.com/google/glog
+    }
+
+    [ -d "protobuf" ] || {
+       git clone --branch v3.9.0 --depth 1 https://github.com/protocolbuffers/protobuf
+    }
+
+    [ -d "libwebsockets" ] || {
+       git clone --branch v3.1-stable --depth 1 https://github.com/warmcat/libwebsockets
+    }
+
+    popd
+}
+
+build_and_install_glog() {
+    REPO_DIR=$1
+    INSTALL_DIR=$2
+    BUILD_DIR=${REPO_DIR}/build_0_3_5
+
+    mkdir -p ${BUILD_DIR}
+    pushd ${BUILD_DIR}
+    cmake .. -DWITH_GFLAGS=off -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
+    make -j${NUM_JOBS}
+    sudo make install
+    popd
+}
+
+build_and_install_protobuf() {
+    REPO_DIR=$1
+    INSTALL_DIR=$2
+    BUILD_DIR=${REPO_DIR}/build_3_9_0
+
+    mkdir -p ${BUILD_DIR}
+    pushd ${BUILD_DIR}
+    cmake ../cmake/ -Dprotobuf_BUILD_TESTS=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
+    make -j${NUM_JOBS}
+    sudo make install
+    popd
+}
+
+build_and_install_websockets() {
+    REPO_DIR=$1
+    INSTALL_DIR=$2
+    BUILD_DIR=${REPO_DIR}/build_3_1_0
+
+    mkdir -p ${BUILD_DIR}
+    pushd ${BUILD_DIR}
+    cmake .. -DLWS_STATIC_PIC=ON -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
+    make -j${NUM_JOBS}
+    sudo make install
+    popd
+}
+
+yes_or_exit() {
+        message=$1
+        while true; do
+                read -p "${message} [Y/n]" yn
+                case $yn in
+                        [Yy]* ) break;;
+                        [Nn]* ) exit;;
+                        * ) echo "Please answer yes or no.";;
+                esac
+        done
+}
+
+setup() {
+        while [[ $# -gt 0 ]]
+        do
+        key="$1"
+        case $key in
+                -y|--yes)
+                answer_yes="True"
+                shift # next argument
+                ;;
+                -h|--help)
+                display_help="True"
+                shift # next argument
+                ;;
+                -b|--builddir)
+                build_dir=$2
+                shift # past argument
+                shift # past value
+                ;;
+                -d|--depsdir)
+                deps_dir=$2
+                shift # past argument
+                shift # past value
+                ;;
+                -i|--depsinstalldir)
+                deps_install_dir=$2
+                shift # past argument
+                shift # past value
+                ;;
+                *)    # unknown option
+                POSITIONAL+=("$1") # save it in an array for later
+                shift # past argument
+                ;;
+        esac
+        done
+        set -- "${POSITIONAL[@]}" # restore positional parameters
+
+        if [[ "${display_help}" == "True" ]]; then
+                print_help
+                exit
+        fi
+        insta
+        ll_required_packages
```

Comment:
this seems broken in 2 lines

Reply:
Something must have happened when I formatted the file, Thank you!